src/js/utils/Select.jsx

```diff
@@ -128,6 +163,24 @@ class Select extends Component {
     return (value && <div className="p-1">{value.label ?? value?.name}</div>);
   }
 
+  mapOptions(values) {
+    return (_.map(values, (value) => {
+      if (typeof value === 'string') {
+        return { value, label: value };
+      }
+      if (value.options) {
+        return { ...value, options: this.mapOptions(value.options) };
+      }
+      if (value && this.props.labelKey && !value.label) {
+        return { ...value, label: value[this.props.labelKey] };
+      }
+      if (value && this.props.valueKey && !value.value) {
+        return { ...value, value: value[this.props.valueKey] };
+      }
+      return value;
```

Comment:
is there a reason you've changed the order of those statements?

Reply:
yep there is,
let me give you a quick example.

Given we have a function with 
``` js
const variable = '';
if (condition A) {
   variable = A;
}
if (condition B) {
   variable = B;
}
if (condition C) {
   variable = C;
}
return variable
```
and if I have conditions **A** = `true` and **C** = `true`, it should return result **C** 

If I want to refactor this function to a _"short-circuit"_ (returning a value once the condition is met)
``` js
if (condition A) {
   return A;
}
if (condition B) {
   return B;
}
if (condition C) {
   return C;
}
```
for the same conditions **A** = `true` and **C** = `true` I get **A**
that's why these `if` statements have to be reordered

Reply:
I'm sorry, I haven't noticed that in the previous version we were not returning, but just re-assigning values